tests/integration/services/test_authentication_service.py

```diff
@@ -0,0 +1,18 @@
+from requests_mock import Mocker
+
+from trailblazer.services.authentication_service.authentication_service import AuthenticationService
+from trailblazer.store.models import User
+
+
+def test_exchange_code(authentication_service: AuthenticationService, user_email: str):
+    # GIVEN an authentication service
+
+    # WHEN exchanging the authorization code
+    token: str = authentication_service.authenticate("auth_code")
```

Comment:
Nice test! It looks a bit finicky to try much in a unit test though (I think this is as far as it goes perhaps). We should probably test this quite extensively so that the tokens actually work as intended as well.

Reply:
I'll have to revise parts of this logic for sure once I start implementing the frontend components